### PR TITLE
builtins: add argn macro

### PR DESCRIPTION
**Pull request by nurmukhametov**
Use it to portably select greater than ninth argument.

### CODE REVIEW

File: builtins/target-avx-utils.ll [hunk 1]
```diff
@@ -294,25 +294,25 @@ define_avgs()
 define(`convert_scale_to_const', `
 
 
- switch i32 %$10, label %default_$1 [ i32 1, label %on_one_$1
+ switch i32 %argn(`10',$@), label %default_$1 [ i32 1, label %on_one_$1
                                       i32 2, label %on_two_$1
```
Comment (dbabokin) [inline builtins/target-avx-utils.ll:298]: Here and in other instances below - need to change alignment of line following the changed line.

**Comment by dbabokin** [comment]
Why do we need this change? I mean what is specific environment that has this problem?

**Comment by nurmukhametov** [comment]
I tried to build in under FreeBSD and their m4 doesn't support two-digits arguments. 

Quote from here that two-digits is GNU extension that is incompatible with POSIX https://www.gnu.org/software/m4/manual/m4.html

> GNU m4 allows the number following the ‘$’ to consist of one or more digits, allowing macros to have any number of arguments. The extension of accepting multiple digits is incompatible with POSIX, and is different than traditional implementations of m4, which only recognize one digit. Therefore, future versions of GNU M4 will phase out this feature. To portably access beyond the ninth argument, you can use the argn macro documented later (see [Shift](https://www.gnu.org/software/m4/manual/m4.html#Shift)).

File: builtins/target-avx-utils.ll (hunk 1 above)
Comment (nurmukhametov) [reply builtins/target-avx-utils.ll]: Fixed

**Pull request merged**